tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/LogsInjectionHelper.cs

```diff
@@ -0,0 +1,157 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Proxies;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Proxies.Pre43;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.LogsInjection
+{
+    /// <summary>
+    /// Helper class to add necessary configuration when logs injection is enabled.
+    /// </summary>
+    internal static class LogsInjectionHelper<TTarget>
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LogsInjectionHelper<TTarget>));
+        private static Type _jsonAttributeType;
+        private static Type _simpleLayoutType;
+
+        /// <summary>
+        ///     Adds necessary configuration elements to inject trace information in logs.
+        /// </summary>
+        /// <param name="loggingConfiguration">The NLog LoggingConfiguration to configure.</param>
+        public static void ConfigureLogsInjection(object loggingConfiguration)
+        {
+            if (loggingConfiguration == null)
+            {
+                return;
+            }
+
+            if (!loggingConfiguration.TryDuckCast<IBasicLoggingConfigurationProxy>(out var loggingConfigurationProxy))
+            {
+                Log.Warning("Failed to DuckCast LoggingConfiguration");
+                return;
+            }
+
+            // this is not available in older versions of NLog (e.g., v2.1 doesn't have JSON support)
+            _jsonAttributeType = Type.GetType("NLog.Layouts.JsonAttribute, NLog", throwOnError: false);
+            if (_jsonAttributeType is null)
+            {
+                return;
+            }
+
+            // this simple layout should exist for all versions
+            _simpleLayoutType = Type.GetType("NLog.Layouts.SimpleLayout, NLog", throwOnError: false);
+            if (_simpleLayoutType is null)
+            {
+                return;
+            }
+
+            ConfigureTargets(loggingConfigurationProxy.ConfiguredNamedTargets);
+        }
+
+        private static void ConfigureTargets(IEnumerable configuredNamedTargets)
+        {
+            foreach (var target in configuredNamedTargets)
+            {
+                if (target.TryDuckCast<ITargetWithLayoutProxy>(out var targetWithLayout))
+                {
+                    var layout = targetWithLayout.Layout;
+
+                    if (layout.TryDuckCast<IJson5LayoutProxy>(out var layoutWithScope))
+                    {
+                        layoutWithScope.IncludeScopePropertiesField = true;
```

Comment:
Do we also need to set the property `IncludeEventProperties=true`? I noticed in our own `DirectSubmissionNLogV5Target` we set both `IncludeScopePropertiesField=true` and `IncludeEventProperties=true`

Reply:
I might be crossing wires on DirectLogsSubmission and LogsInjection stuff, so I'm not sure if this makes sense or not

Reply:
Oh good catch

I'll check to see

Reply:
So looking at this, maybe something just for DirectLogsSubmission as it doesn't seem to impact anything that I can tell for the LogsInjection side of things